python-package/xgboost/spark/core.py

```diff
@@ -685,6 +727,38 @@ def read(cls):
     def _transform(self, dataset):
         raise NotImplementedError()
 
+    def _get_feature_col(self, dataset) -> (list, Optional[list]):
+        """XGBoost model trained with features_cols parameter can also predict
+        vector or array feature type. But first we need to check features_cols
+        and then featuresCol
+        """
+
+        feature_col_names = self.getOrDefault(self.features_cols)
+        features_col = []
+        if (
+            len(feature_col_names)
```

Comment:
This condition looks weird?

```
len(feature_col_names) > 0 >= len(...)
```
is it correct?

Reply:
yes. the pylint required this

Reply:
``` python
all(c in dataset.columns for c in feature_col_names)
```

Reply:
Done, I use another set way to check.